### PR TITLE
Add app identifier config parameter

### PR DESCRIPTION
**Pull request by ufownl**
Relate #478.

### CODE REVIEW

**Comment by Neverlord** [comment]
Wow, thanks for the effort. Looks solid, too. I only have two minor requests:

* the identifier should be a string instead with empty-string as default
  + default strings have 4 bytes on the wire (size with value 0 as `uint32_t`)
  + the `atom` is simply too limited in this case
  + I think there's no reason to forbid names like "my_app_v1.0"
  + only serialized during handshake, so no performance implications anyways
* `basp::instance` does not need the `app_identifier_` field
  + you can access the config directly: `callee_.system().config(). middleman_app_identifier`
  + `basp_broker` should not initialize instance state in `make_behavior`

**Comment by ufownl** [comment]
Tweaks done. And I found that the size of a string in the payload has variant length, the size with value 0 only takes 1 byte.

File: examples/caf-application.ini [hunk 1]
```diff
@@ -39,6 +39,8 @@ relaxed-sleep-duration=10000
 enable-automatic-connections=false
 ; accepted alternative: 'asio' (only when compiling CAF with ASIO)
 network-backend='default'
+; application identifier of this node
+app-identifier=''
```
Comment (Neverlord) [inline examples/caf-application.ini]: `''` is parsed as atom, an empty string is `""` (will fix on my end, just FYI).

**Comment by Neverlord** [comment]
Thank you once more. You rock! :)

**Pull request merged**